src/somef/parser/python_parser.py

```diff
@@ -0,0 +1,387 @@
+"""
+
+def parse_dependency(dependency_str):
+    if not dependency_str:
+        return None, None
+        
+    parts = re.split(r'(>=|<=|==|!=|>|<|~=|\[)', dependency_str, 1)
+    name = parts[0].strip()
+    if len(parts) > 1:
+        version = ''.join(parts[1:])
+    else:
+        version = ""
+
+    version = version.strip("[]() -.,:")
+    
+    return name, version
+
+def parse_url_type(url_label):
+    label = url_label.lower()
+    
+    if "repository" in label or label in ("git", "github", "code", "sourcecode", "source"):
+        return constants.CAT_CODE_REPOSITORY
+    elif "issue" in label or label in ("bug tracker", "tracker", "issues"):
+        return constants.CAT_ISSUE_TRACKER
+    elif "doc" in label or label in ("documentation", "api reference", "reference"):
+        return constants.CAT_DOCUMENTATION
+    elif label == "readme":
+        return constants.CAT_README_URL
+    elif "download" in label:
+        return constants.CAT_DOWNLOAD_URL
+    else:
+        return constants.CAT_RELATED_DOCUMENTATION
+
+def parse_author_string(author_str):
+    if not author_str:
+        return {"name": None, "email": None}
+    
+    email_match = re.search(r'<([^>]+)>', author_str)
+    if email_match:
+        email = email_match.group(1)
+        name = author_str[:email_match.start()].strip()
+        return {"name": name, "email": email}
+    
+    return {"name": author_str, "email": None}
+
+def parse_python_metadata(file_path, metadata_result: Result):
+    try:
+        if Path(file_path).name == "pyproject.toml":
+            with open(file_path, "rb") as f:
+                data = tomli.load(f)
+                
+                project = data.get("project", {})
+                
+                if not project and "tool" in data and "poetry" in data["tool"]:
+                    project = data["tool"]["poetry"]
+                
+                if "name" in project:
+                    metadata_result.add_result(
+                        constants.CAT_NAME,
+                        {
+                            "value": project["name"], 
+                            "type": constants.STRING
+                        },
+                        1,
+                        constants.TECHNIQUE_CODE_CONFIG_PARSER,
+                        file_path
+                    )
+                
+                if "description" in project:
+                    metadata_result.add_result(
+                        constants.CAT_DESCRIPTION,
+                        {
+                            "value": project["description"], 
+                            "type": constants.STRING
+                        },
+                        1,
+                        constants.TECHNIQUE_CODE_CONFIG_PARSER,
+                        file_path
+                    )
+                
+                if "version" in project:
+                    metadata_result.add_result(
+                        constants.CAT_RELEASES,

```

Comment:
Version is not release. We should create a latest version category, or just a version category